net/mwan3/files/lib/mwan3/mwan3.sh

```diff
@@ -1036,11 +1036,23 @@ mwan3_report_iface_status()
 
 	if [ -z "$id" ] || [ -z "$device" ]; then
 		result="offline"
-	elif [ -n "$($IP rule | awk '$1 == "'$((id+1000)):'"')" ] && \
-		     [ -n "$($IP rule | awk '$1 == "'$((id+2000)):'"')" ] && \
-		     [ -n "$($IP rule | awk '$1 == "'$((id+3000)):'"')" ] && \
-		     [ -n "$($IPT -S mwan3_iface_in_$1 2> /dev/null)" ] && \
-		     [ -n "$($IP route list table $id default dev $device 2> /dev/null)" ]; then
+	else
+		error=0
```

Comment:
Could we please move the error check out of the else branch? So we do a check on every run.
So we do not need the `:` in the if statement on line https://github.com/openwrt/packages/pull/13881/commits/8bdbf259f894a9a06df5bc0384e17fc165046dd7#diff-a79526f0a000ba74d1bdaee37e1c9544d0e215cf231a42ab37fa6d29d13c0f64R1054

Reply:
I don't follow what you are proposing.

In master, right now if the `[ -z "$id" ] || [ -z "$device" ]` check passes, then the four checks are not done. Same as in this commit.

In master, it will do the four checks, and if they all pass, it goes on to get more status information. If any fail, it reruns the checks and if they didn't all fail, and then outputs "error", otherwise it outputs offline.

Now, it will  `[ -z "$id" ] || [ -z "$device" ]`, and if it passes, then skip the further checks. If it fails, it will do the checks and encode the results in the error code. This way, if any are failing, it does not need to run the check again. 

We can't do all the checks if we don't have `$id` or `$device`